src/internal/modules/contact-entry/controllers.js

```diff
@@ -0,0 +1,227 @@
+  });
+};
+
+const postSelectContactController = async (request, h) => {
+  const { sessionKey } = request.payload || request.query;
+  const { back } = await sessionHelper.saveToSession(request, sessionKey);
+  const { id, searchQuery, regionId } = request.payload;
+  if (id === null || id === 'new') {
+    return h.redirect(`/contact-entry/new/account-type?sessionKey=${sessionKey}`);
+  } else {
+    const form = forms.handleRequest(selectContact.form(request), request, selectContact.schema);
+    if (!form.isValid) {
+      return h.postRedirectGet(form, '/contact-entry/select-contact', {
+        sessionKey,
+        back: back,
+        regionId,
+        searchQuery
+      });
+    } else {
+      // Contact has been selected. Store the contact ID in yar
+      sessionHelper.saveToSession(request, sessionKey, {
+        newCompany: false,
+        id: id,
+        companyName: request.pre.contactSearchResults.find(x => x.id === id).name
+      });
+      // Redirect to address entry next
+      const queryTail = queryString.stringify({
+        redirectPath: `/contact-entry/new/details/after-address-entry?sessionKey=${sessionKey}`,
+        back: `/contact-entry/new/details?sessionKey=${sessionKey}`
+      });
+      return h.redirect(`/address-entry/postcode?${queryTail}`);
+    }
+  }
+};
+
+const getSelectAccountTypeController = async (request, h) => {
+  const { sessionKey } = request.query;
+  const { accountType, personName, searchQuery } = await sessionHelper.saveToSession(request, sessionKey);
+  return h.view('nunjucks/form', {
+    ...request.view,
+    pageTitle: 'Select the account type',
+    form: sessionForms.get(request, selectAccountType.form(request, accountType, personName || searchQuery))
+  });
+};
+
+const postSelectAccountTypeController = async (request, h) => {
+  const { sessionKey } = request.payload || request.query;
+  const { accountType, personName } = request.payload;
+  const form = forms.handleRequest(
+    selectAccountType.form(request),
+    request,
+    selectAccountType.schema
+  );
+  // If form is invalid, redirect user back to form
+  if (!form.isValid) {
+    return h.postRedirectGet(form, '/contact-entry/new/account-type', {
+      sessionKey
+    });
+  } else {
+    // Contact has been selected. Store the contact account type in yar
+    sessionHelper.saveToSession(request, sessionKey, { newCompany: true, accountType, personFullName: personName, companyName: personName });
+
+    // Proceed to the next stage
+    if (accountType === 'organisation') {
+      return h.redirect(`/contact-entry/new/details?sessionKey=${sessionKey}`);
+    } else {
+      // Goes to the address entry workflow
+      const queryTail = queryString.stringify({
+        redirectPath: `/contact-entry/new/details/after-address-entry?sessionKey=${sessionKey}`,
+        back: `/contact-entry/new/details?sessionKey=${sessionKey}`
+      });
+      return h.redirect(`/address-entry/postcode?${queryTail}`);
```

Comment:
Here you should redirect back to the invoice accounts select address for an existing company passing back the selected company id as in the example below; 

return h.redirect(`/invoice-accounts/create/${regionId}/${companyId}/select-address?agentId=selectedCompanyId`);